### ISSUE TITLE
Workbench Crashes When Data is Replaced with the Show Data Window Open

### ISSUE THREAD

**Issue by DannyHindson** [mantidproject/mantid#33885]
Mantid will crash from any kind of workspace replacement from a load algorithm (there may be other causes, but this is the most consistent). 

**Reproduction:**
1. Load any workspace
2. Right click -> show data
3. Load the same workspace with the same name to overwrite the data
4. UnEx occurs
5. Workbench may continue to work, but may hang, and minimising on mac causes a crash to desktop. 

Stack trace:
```
Traceback (most recent call last):
File "/opt/mantidworkbenchnightly/lib/python3.8/site-packages/mantidqt/widgets/workspacedisplay/matrix/presenter.py", line 71, in action_replace_workspace
if self.model.workspace_equals(workspace_name):
File "/opt/mantidworkbenchnightly/lib/python3.8/site-packages/mantidqt/widgets/workspacedisplay/matrix/model.py", line 52, in workspace_equals
return workspace_name == self._ws.name()
RuntimeError: Variable invalidated, data has been deleted.
```

**Comment by DanielMurphy22** [comment]
Not a problem on Release 6.3

**Comment by DanielMurphy22** [comment]
If I comment out this Line 71 and always run the contents of the if statement then the behaviour seems to be fixed. I don't quite have the understanding yet so this is treating a symptom not the cause: https://github.com/mantidproject/mantid/blob/main/qt/python/mantidqt/mantidqt/widgets/workspacedisplay/matrix/presenter.py#L71

**Comment by DanielMurphy22** [comment]
This is the first bad commit: https://github.com/mantidproject/mantid/commit/5a8567449036e7c9253be45821a246444e8d9946

**Comment by DanielMurphy22** [comment]
This bug does not exist for ShowData on TableWorkspaces

**Comment by DanielMurphy22** [comment]
Fixed by #33916
